src/bitcoin-cli.cpp

```diff
@@ -235,7 +235,12 @@ UniValue CallRPC(const std::string& strMethod, const UniValue& params)
     assert(output_buffer);
     evbuffer_add(output_buffer, strRequest.data(), strRequest.size());
 
-    int r = evhttp_make_request(evcon.get(), req.get(), EVHTTP_REQ_POST, "/");
+    // check if we should use a special wallet endpoint
+    std::string endpoint = "/";
+    if (GetArg("-wallet", "") != "") {
```

Comment:
In commit "Add -wallet endpoint support to bitcoin-cli":

You should update `HelpMessageCli` to mention the new argument.

Since I started making a similar change (but didn't get very far) you could steal my help string:

```diff
strUsage += HelpMessageOpt("-wallet=<file>", _("Send RPC for non-default wallet on RPC server (argument is wallet filename in bitcoind directory)"));
```